components/header-bar/src/profile/profile-menu.js

```diff
@@ -69,6 +69,20 @@ const ProfileContents = ({ name, email, avatarUrl, helpUrl }) => {
                             baseUrl,
                             'dhis-web-commons-security/logout.action'
                         )}
+                        // NB: By MenuItem implementation, this callback
+                        // overwrites default navigation behavior but maintains
+                        // the href attribute
+                        onClick={async () => {
+                            if (onLogout) {
+                                await onLogout()
+                            }
+                            window.location.assign(
+                                joinPath(
+                                    baseUrl,
+                                    'dhis-web-commons-security/logout.action'
+                                )
+                            )
+                        }}
```

Comment:
await `onLogout` and navigate

Reply:
This navigation method is limited when it is done in an `iframe` -- is there a way to emulate exactly what happens when clicking on an `a` with an `href`?

Reply:
I'm not sure myself... maybe just send a `fetch`?

Reply:
It seems like `fetch` also needs to be followed up by a `window.location` type action too